engine/src/main/java/org/hibernate/search/engine/environment/thread/impl/EmbeddedThreadProvider.java

```diff
@@ -31,13 +30,8 @@ public String createThreadName(String prefix, int threadNumber) {
 	}
 
 	@Override
-	@SuppressForbiddenApis(reason = "It's unclear how we will handle this without the security manager;"
-			+ " we'll see when the security manager actually gets removed from the JDK")
 	public ThreadFactory createThreadFactory(String prefix) {
-		@SuppressWarnings("removal")
-		SecurityManager s = System.getSecurityManager();
-		@SuppressWarnings({ "removal", "deprecation" })
-		ThreadGroup group = ( s != null ) ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
```

Comment:
Should we have some ugly static reflection code that detects whether `System.getSecurityManager` exists, and depending on that does the right thing?

I have no idea of the impact of this change on existing apps... Don't get me wrong, I've fine with this changing behavior, but if it does, we'll need to postpone the change to a major version of Search... But maybe you're planning on having one anyway for when ORM 7.0 is out?

Reply:
hmm so maybe let's do the reflection thing and keep the `Security Exception`s in place and get this merged into the current main (7.2) and then for the next major we remove all that ?

Reply:
Works for me. Though at this point you may as well open another PR and keep this one for the next major :)

BTW, do we have a version of the JDK we could test the reflection hack against? I think everything still works on JDK 23, doesn't it? I'll do a JDK upgrade on Jenkins just in case.

Reply:
I'm looking at 23.ea.18-open, and it seems like it still has SM in place...